### PR TITLE
Add missing response headers

### PR DESCRIPTION
**Pull request by tschuehly**
As discussed in https://github.com/wimdeblauwe/htmx-spring-boot/issues/105 this is the work of @xhaggi rebased on the current main and fixed to align with the current version

source: https://github.com/xhaggi/htmx-spring-boot/tree/missing-response-headers

### CODE REVIEW

**Comment by xhaggi** [comment]
I would definitely prefer if you cherry pick the commits so that you have one per annotation and not to lose the author of the commits. @wimdeblauwe or do you have a different opinion on this?

**Comment by tschuehly** [comment]
@xhaggi 
Well then I need to solve merge conflicts for each step, it already took two hours as it is now.

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxLocation.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    String handler() default "";
+    /**
+     * The target to swap the response into.
+     */
+    String target() default "";
+    /**
+     * How the response will be swapped in relative to the target
+     */
+    String swap() default "";
```
Comment (wimdeblauwe) [inline htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxLocation.java:54]: Can't we use `HxSwapType` here ?

**Comment by wimdeblauwe** [comment]
> I would definitely prefer if you cherry pick the commits so that you have one per annotation and not to lose the author of the commits. @wimdeblauwe or do you have a different opinion on this?

I like small commits for sure, but I think this is a nice whole. I don't see much need for separate commits for each annotation. That said, maybe @tschuehly can do a force commit with an added line in the commit message using `Co-authored-by` (https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) to acknowledge all the work you have done on it.

**Comment by xhaggi** [comment]
> Well then I need to solve merge conflicts for each step, it already took two hours as it is now.

@tschuehly I really appreciate your work, but I like to stick to separate commits and the original work I did. That's why I did it on my own and created a separate PR #118  for it. I hope this is OK for you? Maybe you should have asked in #105 if I could do the rebase and a PR, I certainly would have done it. BTW it only took me a few minutes.

**Comment by tschuehly** [comment]
Well now it doesn't matter 😁

**Pull request closed without merging**